File: bencoding/Java/BencodingLexerBase.java
import org.antlr.v4.runtime.CharStream;
import org.antlr.v4.runtime.Lexer;

public abstract class BencodingLexerBase extends Lexer {

  private int stringLength = 0;

  protected BencodingLexerBase(CharStream input) {
    super(input);
  }

  protected void setStringLength() {
    String tokenText = super.getText();
    this.stringLength = Integer.parseInt(tokenText.substring(0, tokenText.length() - 1));
  }

  protected boolean consumeStringChars() {
    this.stringLength--;
    return this.stringLength > -1;
  }
}

File: bencoding/CSharp/BencodingLexerBase.cs
using System;
using System.Collections.Generic;
using System.IO;
using Antlr4.Runtime;

public abstract class BencodingLexerBase : Lexer
{
  private int _stringLength = 0;

  public BencodingLexerBase(ICharStream input)
    : base(input)
  {
  }

  public BencodingLexerBase(ICharStream input, TextWriter output, TextWriter errorOutput)
    : base(input, output, errorOutput)
  {
  }

  public void setStringLength() => this._stringLength = int.Parse(this.Text[..^1]);

  public bool consumeStringChars() => (--this._stringLength) > -1;
}
